### PR TITLE
[DOCS] Adds feature processor conceptual docs to ML book

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds conceptual docs on custom feature processors to the ML book. The following encoding techniques are covered:
* frequency encoding
* multi encoding
* ngram encoding
* one-hot encoding
* target mean encoding.

### Preview

[Feature processors](https://stack-docs_1540.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-feature-processors.html)

### Notes

The section about multi encoding must be backported only to 7.12.

### CODE REVIEW

File: docs/en/stack/ml/df-analytics/ml-dfa-feature-processors.asciidoc [hunk 1]
```diff
@@ -0,0 +1,97 @@
+[role="xpack"]
+[[ml-feature-processors]]
+= {dfanalytics-cap} feature processors
+++++
+<titleabbrev>Feature processors</titleabbrev>
+++++
+
+beta::[]
+
+Feature processors enable you to extract process features from raw document
```
Comment (benwtrent) [inline docs/en/stack/ml/df-analytics/ml-dfa-feature-processors.asciidoc:10]:
```suggestion
Feature processors enable you to extract process features from the raw document 
```

File: docs/en/stack/ml/df-analytics/ml-dfa-feature-processors.asciidoc [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+* <<ngram-encoding>>
+* <<one-hot-encoding>>
+* <<target-mean-encoding>>
+
+
+[[frequency-encoding]]
+== Frequency encoding
+
+Frequency encoding takes into account how many times a given categorical feature 
+is present in relation to a particular value in the data set. The more
```
Comment (benwtrent) [inline docs/en/stack/ml/df-analytics/ml-dfa-feature-processors.asciidoc:34]: This has to do with the value of the field that is being encoded. Not just any particular value in the data set. We should make the relationship clearer.

File: docs/en/stack/ml/df-analytics/ml-dfa-feature-processors.asciidoc [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+bigram encoding (ngram of size 1 and 2) and goes to the string length of 3._
+
+
+[[one-hot-encoding]]
+== One hot encoding
+
+One hot encoding transforms categorical values into numerical ones by assigning 
+vectors to each category. The vector represents whether the corresponding 
+feature is present (1) or not present (0) at the given value, so the encoding 
+method maps the different categorical features to the numerical values.
```
Comment (benwtrent) [inline docs/en/stack/ml/df-analytics/ml-dfa-feature-processors.asciidoc:76]:
```suggestion
One hot encoding transforms from categorical to numerical values by assigning 
a 1 or a 0 to each category. The value represents whether the corresponding 
feature value is present (1) or not present (0), so the encoding 
method maps the different categorical features to the numerical values.
```

The word `vectors` seems strange to me. Seems like saying `assigning 1 or 0` is much simpler.

**Review by benwtrent — APPROVED** [review]

**Review by lcawl — APPROVED** [review]
LGTM, thanks for the clarifications!

**Pull request merged**